### PR TITLE
Added error message when adding an already assigned tag to a device

### LINKED ISSUE #989: No Error When Trying To Add Same Tag To Device Multiple Times
Kapua does not return any kind of error when trying to add same tag multiple times to certain device. Instead info box is shown, saying: **_Tag added_**. 
Nevertheless tag is not added twice (so basically only info message has to be replaced with an error message) - look at the screenshot. 

Testflow: 
1. Login as kapua-sys
2. Go to Device, connect or add a device
3. Go to its tags and try to add sam tag twice
**Kapua does not add tag twice, but the info box says that the tag has been added (_Tag added_).**

screen shot 2017-11-16 at 14 49 05

Browsers used:
- Google Chrome (Version 62.0.3202.94 (Official Build) (64-bit))
- Safari (Version 11.0.1 (13604.3.5))
- Mozilla Firefox (57.0 (64-bit))
- Opera (Version 49.0.2725.34)

### PR DESCRIPTION
**Pull request by ct-goranpalibrk**
Signed-off-by: CT\pgoran <goran.palibrk@comtrade.com>

### CODE REVIEW

File: console/module/device/src/main/resources/org/eclipse/kapua/app/console/module/device/client/messages/ConsoleDeviceMessages.properties [hunk 1]
```diff
@@ -337,6 +337,7 @@ dialogDeviceTagAddFieldTagEmptyTextNoTags=No tags found...
 dialogDeviceTagAddFieldTagLoadingError=Unable to load tag list: {0}
 dialogDeviceTagAddConfirmation=Tag added
 dialogDeviceTagAddError=Error when adding tag: {0}
+dialogDeviceTagAddErrorSameId=Tag with selected name already added!
```
Comment (Coduz) [inline console/module/device/src/main/resources/org/eclipse/kapua/app/console/module/device/client/messages/ConsoleDeviceMessages.properties]: Please change this in: 
"Tag {0} was already added"

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/server/GwtDeviceServiceImpl.java [hunk 2]
```diff
@@ -453,13 +454,21 @@ public void addDeviceTag(GwtXSRFToken xsrfToken, String scopeIdString, String de
             Device device = drs.find(scopeId, deviceId);
 
             Set<KapuaId> tagIds = device.getTagIds();
```
Comment (Coduz) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/server/GwtDeviceServiceImpl.java]: Please throw an exception when the same tag was already added.

File: console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/client/device/tag/DeviceTagAddDialog.java [hunk 3]
```diff
@@ -55,12 +55,17 @@ public void submit() {
...
 
-        GWT_DEVICE_SERVICE.addDeviceTag(xsrfToken, selectedDevice.getScopeId(), selectedDevice.getId(), tagId, new AsyncCallback<Void>() {
+        GWT_DEVICE_SERVICE.addDeviceTag(xsrfToken, selectedDevice.getScopeId(), selectedDevice.getId(), tagId, new AsyncCallback<Boolean>() {
 
             @Override
-            public void onSuccess(Void gwtAccessPermission) {
+            public void onSuccess(Boolean gwtAccessPermission) {
                 exitStatus = true;
-                exitMessage = MSGS.dialogDeviceTagAddConfirmation();
+                if (gwtAccessPermission) {
```
Comment (Coduz) [inline console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/client/device/tag/DeviceTagAddDialog.java]: After throwing the exception on the server side, manage the error message in `onFailure` method, showing the message related to the duplicate tag when it happens or a generic message if other exception occurs

**Comment by Coduz** [comment]
I'll change the title of the PR to a more meaningful one.

This PR fixes #989

**Comment by LeoNerdoG** [comment]
I have verified that this PR resolves issue #989 .

**Pull request merged**